### PR TITLE
Added contributors and sponsors

### PR DESCRIPTION
**Pull request by jcohenadad**
This PR adds features to the documentation:
- Added contributors and sponsors
- Added Sections on the left TOC
- Minor reorganization

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+.. note:: This website is under construction.
+
+``ivadomed`` is an integrated framework for medical image analysis with deep
+learning. The name is a portmanteau between *IVADO* (The `Institute for data
+valorization <https://ivado.ca/en/>`_) and *Medical*.
+
+The purpose of the ``ivadomed`` project is to:
+
+- Provide researchers with an open-source framework for training deep learning models for applications in medical imaging;
```
Comment (ieee8023) [inline docs/source/index.rst:12]: Maybe list the types of models or narrow it down somehow.

File: docs/source/technical_features.md [hunk 2]
```diff
@@ -32,15 +17,15 @@ positive and negative samples, negative samples being under-weighted dynamically
 at each epoch. During the second step, the CNN is fine-tuned on the realistic
 (i.e. class-imbalanced) dataset.
 
-### Mixup
+## Mixup
 [Mixup](https://arxiv.org/pdf/1710.09412.pdf) is a data augmentation technique,
```
Comment (ieee8023) [inline docs/source/technical_features.md:21]: I could site the abstract link and not the pdf: https://arxiv.org/abs/1710.09412

File: docs/source/technical_features.md [hunk 3]
```diff
@@ -51,11 +36,11 @@ the prior knowledge about the subjective lesion borders is then leveraged to the
 
 ![Figure Data Augmentation on lesion ground truths](https://github.com/neuropoly/ivado-medical-imaging/raw/master/images/dilate-gt.png)
 
-### Network architectures
+## Network architectures
 - [U-net](https://arxiv.org/pdf/1505.04597.pdf)
```
Comment (ieee8023) [inline docs/source/technical_features.md:40]: Maybe also reference what files these implementations are in?

File: docs/source/technical_features.md (hunk 2 above)
Comment (jcohenadad) [reply docs/source/technical_features.md]: fixed in 93868ce7ffeba5a35b420e280f2d71716e1376f4

File: docs/source/technical_features.md (hunk 3 above)
Comment (jcohenadad) [reply docs/source/technical_features.md:40]: what do you mean by "what files"? do you mean that we should add a link that points to the actual files in the code that implement these models? If so, this is out of the scope of this PR (if that's what you meant i'll open an issue)

Comment (ieee8023) [reply docs/source/technical_features.md:40]: Ya, that is what I meant. I don't know if it needs to be done, just throwing out the idea.

Comment (jcohenadad) [reply docs/source/technical_features.md:40]: great idea-- issue opened https://github.com/neuropoly/ivado-medical-imaging/issues/270

**Pull request merged**